src/site/content/en/blog/compute-pressure/index.md

```diff
@@ -0,0 +1,284 @@
+
+When a device's CPU utilization gets high, the device may increase clock speeds
+across its CPU cores, in an attempt to meet the demand for computing power. As
+the speeds exceed the base clock speed, the elevated power consumption increases
+the CPU's temperature. At some point, the device enters a
+[thermal throttling regime](https://en.wikipedia.org/wiki/Dynamic_frequency_scaling),
+where the CPU clock speed is reduced, in order to bring the temperature down.
+
+As with CPU utilization, values from `0.0` (minimal) to `1.0` (boosted) are used
+to normalize clock speeds. A value of `0.5` represents the base clock speed for
+a given CPU.
+
+## Using the API
+
+The following code creates a `ComputePressureObserver`, configured with specific
+triggering thresholds, along with an associated callback function that may be
+invoked when a threshold is crossed. The callback is intentionally rate-limited,
+with the current implementation maxing out at one invocation per second for a
+page in the foreground, and once per ten seconds for background pages.
+
+Inside the callback function, your web application can respond to the changing
+utilization and clock speed levels by enabling or disabling CPU-intensive
+functionality.
+
+```javascript
+function callback(update) {
+  if (update.cpuSpeed > 0.5) {
+    // The CPU is running at faster than base speed.
+  } else {
+    // The CPU is running at normal or reduced speed.
+  }
+
+  if (update.cpuUtilization >= 0.9) {
+    // CPU utilization is over 90%.
+  } else if (update.cpuUtilization >= 0.75) {
+    // CPU utilization is over 75%.
+  } else if (update.cpuUtilization >= 0.5) {
+    // CPU utilization is over 50%.
+  } else {
+    // CPU utilization is under 50%.
+  }
+}
+
+const observer = new ComputePressureObserver(callback, {
+  // Thresholds divide the interval [0.0 .. 1.0] into ranges.
+  cpuUtilizationThresholds: [0.5, 0.75, 0.9],
+  // The minimum clock speed is 0.0, and the maximum speed is 1.0.
+  // 0.5 represents the base clock speed.
+  cpuSpeedThresholds: [0.5],
+});
+
+// Begin observing changes and triggering callbacks:
+observer.observe();
+
+// ...later on, to stop observing changes:
+observer.unobserve();
+```
+
+### Exposing limited information via thresholds and buckets
+
+One of the key API design goals is to provide enough information for an
+application to make useful decisions based on compute pressure, while not
+exposing information that might
+[harm a user's privacy](https://github.com/oyiptong/compute-pressure/blob/main/security-privacy-self-assessment.md).
+
+With this in mind, the API reports back normalized, unitless values between
+`0.0` and `1.0` instead of more identifiable values like clock speeds in GHz.
+The values passed to the callback fall into one of the "buckets" defined by the
+thresholds used to configure the observer. The actual value will be the midpoint
+between the upper and lower bounds for the bucket—in other words, they will be
```

Comment:
```suggestion
between the upper and lower bounds for the bucket. In other words, they will be
```

Reply:
One day you'll let me sneak an em dash through 😄